### PR TITLE
xoshiro-cpp: conan v2 support

### PR DESCRIPTION
**Pull request by robomics**
Specify library name and version:  **xoshiro-cpp/all**

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by robomics** [comment]
ready for review :)

File: recipes/xoshiro-cpp/all/conanfile.py [hunk 1]
```diff
@@ -24,45 +28,47 @@ def _minimum_compilers_version(self):
...
+             dst=os.path.join(self.package_folder, "include/xoshiro-cpp/"))
+        copy(self, "LICENSE", src=self.source_folder,
+             dst=os.path.join(self.package_folder, "licenses"))
 
     def package_info(self):
-        self.cpp_info.names["cmake_find_package"] = "xoshiro-cpp"
-        self.cpp_info.names["cmake_find_package_multi"] = "xoshiro-cpp"
+        self.cpp_info.set_property("cmake_file_name", "xoshiro-cpp")
+        self.cpp_info.set_property(
+            "cmake_target_name", "xoshiro-cpp::xoshiro-cpp")
```
Comment (memsharded) [inline recipes/xoshiro-cpp/all/conanfile.py:72]: There are already the defaults, they might be dropped (not critical, can be there, no need to change it)

**Review by memsharded — APPROVED** [review]
Thanks!

**Review by SSE4 — APPROVED** [review]

File: recipes/xoshiro-cpp/all/conanfile.py [hunk 2]
```diff
@@ -24,45 +28,47 @@ def _minimum_compilers_version(self):
...
                 raise ConanInvalidConfiguration(msg)
         except KeyError:
             msg = (
-                "{} recipe lacks information about the {} compiler, "
-                "support for the required C++{} features is assumed"
-            ).format(self.name, compiler, self._minimum_cpp_standard)
+                f"{self.ref} recipe lacks information about the {compiler} compiler, "
+                f"support for the required C++{self._minimum_cpp_standard} features is assumed"
+            )
             self.output.warn(msg)
```
Comment (SpaceIm) [inline recipes/xoshiro-cpp/all/conanfile.py:57]: it's invalid in conan v2

**Pull request merged**